java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -152,130 +130,358 @@ private void execute(String signature, SquidClassLoader classLoader) {
+        break;
+      case DUP2_X1:
+        pop = popStack(3, instruction.opcode);
+        programState = pop.state
+          .stackValue(pop.values.get(1))
+          .stackValue(pop.values.get(0))
+          .stackValue(pop.values.get(2))
+          .stackValue(pop.values.get(1))
+          .stackValue(pop.values.get(0));
+        break;
+      case DUP2_X2:
+        pop = popStack(4, instruction.opcode);
+        programState = pop.state
+          .stackValue(pop.values.get(1))
+          .stackValue(pop.values.get(0))
+          .stackValue(pop.values.get(3))
+          .stackValue(pop.values.get(2))
+          .stackValue(pop.values.get(1))
+          .stackValue(pop.values.get(0));
+        break;
+      case SWAP:
+        pop = popStack(2, instruction.opcode);
+        programState = pop.state.stackValue(pop.values.get(0)).stackValue(pop.values.get(1));
+        break;
+      case IADD:
+      case LADD:
+      case FADD:
+      case DADD:
+      case ISUB:
+      case LSUB:
+      case FSUB:
+      case DSUB:
+      case IMUL:
+      case LMUL:
+      case FMUL:
+      case DMUL:
+      case IDIV:
+      case LDIV:
+      case FDIV:
+      case DDIV:
+      case IREM:
+      case LREM:
+      case FREM:
+      case DREM:
+      case ISHL:
+      case LSHL:
+      case ISHR:
+      case LSHR:
+      case IUSHR:
+      case LUSHR:
+        pop = popStack(2, instruction.opcode);
+        SymbolicValue sv1 = constraintManager.createSymbolicValue(instruction);
+        programState = pop.state.stackValue(sv1).addConstraint(sv1, ObjectConstraint.NOT_NULL);
+        break;
+      case INEG:
+      case LNEG:
+      case FNEG:
+      case DNEG:
+        pop = popStack(1, instruction.opcode);
+        sv = constraintManager.createSymbolicValue(instruction);
+        programState = pop.state.stackValue(sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case IAND:
+      case LAND:
+      case IOR:
+      case LOR:
+      case IXOR:
+      case LXOR:
+        pop = popStack(2, instruction.opcode);
+        sv = constraintManager.createBinarySymbolicValue(instruction, pop.valuesAndSymbols);
+        programState = pop.state.stackValue(sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case IINC:
+        int index = instruction.operand;
+        SymbolicValue existing = programState.getValue(index);
+        Preconditions.checkNotNull(existing, "Local variable " + index + " not found");
+        sv = constraintManager.createSymbolicValue(instruction);
+        programState = programState.put(index, sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case I2L:
+      case I2F:
+      case I2D:
+      case L2I:
+      case L2F:
+      case L2D:
+      case F2I:
+      case F2L:
+      case F2D:
+      case D2I:
+      case D2L:
+      case D2F:
+      case I2B:
+      case I2C:
+      case I2S:
+        // TODO SONARJAVA-2508 handle value conversion
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)